File: app/ui/src/main/java/com/fsck/k9/activity/K9Activity.java
package com.fsck.k9.activity;


import android.os.Bundle;

import androidx.annotation.LayoutRes;
import androidx.appcompat.app.AppCompatActivity;
import androidx.appcompat.widget.Toolbar;

import com.fsck.k9.ui.R;
import com.fsck.k9.ui.ThemeManager;


public abstract class K9Activity extends AppCompatActivity {
    private final K9ActivityCommon base = new K9ActivityCommon(this, ThemeType.DEFAULT);

    public ThemeManager getThemeManager() {
        return base.getThemeManager();
    }

    @Override
    public void onCreate(Bundle savedInstanceState) {
        base.preOnCreate();
        super.onCreate(savedInstanceState);
    }

    @Override
    protected void onResume() {
        base.preOnResume();
        super.onResume();
    }

    protected void setLayout(@LayoutRes int layoutResId) {
        setContentView(layoutResId);
        Toolbar toolbar = findViewById(R.id.toolbar);
        if (toolbar == null) {
            throw new IllegalArgumentException("K9 layouts must provide a toolbar with id='toolbar'.");
        }
        setSupportActionBar(toolbar);
    }
}

File: app/ui/legacy/src/main/java/com/fsck/k9/activity/K9Activity.kt
package com.fsck.k9.activity

import android.os.Bundle
import androidx.annotation.LayoutRes
import androidx.appcompat.app.AppCompatActivity
import androidx.appcompat.widget.Toolbar
import com.fsck.k9.ui.R
import com.fsck.k9.ui.ThemeManager

abstract class K9Activity : AppCompatActivity() {
    private val base = K9ActivityCommon(this, ThemeType.DEFAULT)

    val themeManager: ThemeManager
        get() = base.themeManager

    override fun onCreate(savedInstanceState: Bundle?) {
        base.preOnCreate()
        super.onCreate(savedInstanceState)
    }

    override fun onResume() {
        base.preOnResume()
        super.onResume()
    }

    protected fun setLayout(@LayoutRes layoutResId: Int) {
        setContentView(layoutResId)
        val toolbar = findViewById<Toolbar>(R.id.toolbar)
            ?: error("K9 layouts must provide a toolbar with id='toolbar'.")

        setSupportActionBar(toolbar)
    }
}
